### ISSUE TITLE
Linter running but errors not showing

### ISSUE THREAD

**Issue by medwatt** [dense-analysis/ale#3340]
## Information

**VIM version** 
Neovim v0.5.0-675
Operating System:  Windows 10

## What went wrong

I installed the ale plugin with vim-plug. I have flake8 (python linter) installed. Everything worked just as expected. I made some error somewhere and an error symbol was immediately displayed. For some reason, I see no errors are displayed when editing VHDL and Verilog files, even though they are supported.

## Reproducing the bug

I have already established that everything works fine for python files. I want to be able to see errors when I edit VHDL and Verilog files. I currently have modelsim, ghdl, vivado, and iverilog. These are all tools that are supported for VHDL and Verilog. The installation directories are already included in PATH and I can call them from the command line. In fact, when I use them with syntastic, they display errors, but only after I save the file. I don't know why I am not seeing any errors when they are used under ALE.

This is the only settings I have added for ALE:

```vim
let g:ale_linters = {
\   'verilog': ['vlog', 'iverilog' ],
\   'verilog_systemverilog': ['vlog', 'iverilog' ],
\   'vhdl': ['vcom', 'ghdl'],
\   'python': ['flake8'],
\}
```

### :ALEInfo

```vim
 Current Filetype: vhdl
Available Linters: ['ghdl', 'hdl-checker', 'vcom', 'xvhdl']
  Enabled Linters: ['ghdl', 'vcom']
 Suggested Fixers: 
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_vhdl_ghdl_executable = 'ghdl'
let g:ale_vhdl_ghdl_options = '--std=08'
let g:ale_vhdl_vcom_executable = 'vcom'
let g:ale_vhdl_vcom_options = '-2008 -quiet -lint'
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'verilog': ['vlog', 'iverilog'], 'vhdl': ['vcom', 'ghdl'], 'verilog_systemverilog': ['vlog'], 'python': ['flake8']}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) ghdl
(finished - exit code 1) 'cmd /s/c "ghdl -s  --std=08 C:\Users\MOHAME~1\AppData\Local\Temp\nvim6B2qey\0\dual_port_ram.vhd"'

<<<OUTPUT STARTS>>>
C:\Users\MOHAME~1\AppData\Local\Temp\nvim6B2qey\0\dual_port_ram.vhd:3:25: missing ";" at end of use clause
<<<OUTPUT ENDS>>>

(executable check - success) vcom
(finished - exit code 2) 'cmd /s/c "vcom  -2008 -quiet -lint -work C:\Users\MOHAME~1\AppData\Local\Temp\nvim6B2qey\1 C:\Users\MOHAME~1\AppData\Local\Temp\nvim6B2qey\2\dual_port_ram.vhd"'

<<<OUTPUT STARTS>>>
** Error: C:\Users\MOHAME~1\AppData\Local\Temp\nvim6B2qey\2\dual_port_ram.vhd(5): near "entity": (vcom-1576) expecting '[' or '.' or '(' or '\''.
<<<OUTPUT ENDS>>>

```

**Comment by medwatt** [comment]
Surely, there's someone here that can help ?

**Comment by medwatt** [comment]
Thanks to azvanderpas for pointing out that the issue has to do with the regex not recognizing windows path format. A very simple solution is to open the linters files and adding the two symbols :~ in the regular expressions in the offended files. For instance, in the ghdl.vim file, the regular expression pattern is :

```
^[a-zA-Z0-9\-\.\_\/ ]\+:\(\d\+\):\(\d\+\):\(.*\)
```
All we need to do is add :~ and get:
```
^[a-zA-Z0-9\-\.\_\/:~ ]\+:\(\d\+\):\(\d\+\):\(.*\)
```

The same can be done for the vhdl and verilog files.
